preparation-kits/one-week/d7-no_prefix_set.py

```diff
@@ -0,0 +1,25 @@
+def validate_set(strings):
+    accepted_strings = set()
+    invalid_strings = set()
+    
+    def print_bad_set(string):
+        print("BAD SET")
+        print(string)
+
+    for string in strings:
+        if string in invalid_strings:
+            print_bad_set(string)
+            return
+        
+        substring = ''
+        for char in string:
+            substring += char
+            if substring in accepted_strings:
+                print_bad_set(string)
+                return
+            else:
+                invalid_strings.add(substring)
```

Comment:
No need for the else if the if returns the code.
```suggestion
            invalid_strings.add(substring)
```